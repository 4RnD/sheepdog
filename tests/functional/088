#!/bin/bash

# Test data integrity after changing my address (sheep -y <myaddr>)

. ./common

MY_ADDR=`hostname -I | awk '{print $1}'`
ORIG_SHEEP_OPTIONS=$SHEEP_OPTIONS
NEW_SHEEP_OPTIONS="-n -y $MY_ADDR -l level=debug"

echo "1. start sheep with my address"
SHEEP_OPTIONS=$NEW_SHEEP_OPTIONS
_start_sheep 0
_wait_for_sheep 1
echo "----- cluster format ------"
_cluster_format -c 1
echo "----- cluster info -----"
$DOG cluster info | _filter_cluster_info | _filter_ip
echo "----- vdi create -----"
$DOG vdi create test 40M -P
echo "----- vdi list -----"
_vdi_list
echo "----- cluster shutdown -----"
$DOG cluster shutdown
_wait_for_sheep_stop

echo "2. start sheep with default"
SHEEP_OPTIONS=$ORIG_SHEEP_OPTIONS
_start_sheep 0
_wait_for_sheep 1
echo "----- cluster info -----"
$DOG cluster info | _filter_cluster_info | _filter_ip
echo "----- node info -----"
$DOG node info
echo "----- cluster shutdown -----"
$DOG cluster shutdown
_wait_for_sheep_stop

echo "3. start sheep with my address again"
SHEEP_OPTIONS=$NEW_SHEEP_OPTIONS
_start_sheep 0
_wait_for_sheep 1
echo "----- cluster info -----"
$DOG cluster info | _filter_cluster_info | _filter_ip
echo "----- vdi list -----"
_vdi_list
echo "----- cluster shutdown -----"
$DOG cluster shutdown
_wait_for_sheep_stop
