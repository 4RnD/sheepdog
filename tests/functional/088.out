QA output created by 088
1. start sheep with my address
----- cluster format ------
using backend plain store
----- cluster info -----
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version
DATE      1 [IP:7000]
----- vdi create -----
----- vdi list -----
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   40 MB   40 MB  0.0 MB DATE   7c2b25      1                22
----- cluster shutdown -----
2. start sheep with default
----- cluster info -----
Cluster status: Waiting for other nodes to join cluster

Cluster created at DATE

Epoch Time           Version
DATE      1 [IP:7000]
----- node info -----
Response's result: Waiting for other nodes to join cluster
Cannot get information from any nodes
Id	Size	Used	Avail	Use%
----- cluster shutdown -----
3. start sheep with my address again
----- cluster info -----
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version
DATE      1 [IP:7000]
----- vdi list -----
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   40 MB   40 MB  0.0 MB DATE   7c2b25      1                22
----- cluster shutdown -----
